Samc spring 6

SamuelConnor opened the pull request.

SamuelConnor committed:
Story 5:
-Booking CRUD

SamuelConnor committed:
Story 6:
-Endpoint added for finding todays bookings.

Kieransullivan-imanage commented on src/main/java/com/example/demo/services/BookingService.java:
```diff
@@ -0,0 +1,78 @@
+
+    public BookingEntity findBookingById(Long id) {
+
+        if(!bookingEntityRepository.existsById(id)) {
```
double braces.

Kieransullivan-imanage commented on src/main/java/com/example/demo/services/BookingService.java:
```diff
@@ -0,0 +1,78 @@
+        if(!bookingEntityRepository.existsById(id)) {
+            { throw new BookingNotFoundException(); }
+        }
+        return bookingEntityRepository.findById(id).get();
```
If you want to get an entity and throw an exception if it didn't;t exist, rather than perform two repo operations you can do it in one with findById(id).orElseThrow(Exception:new)

Kieransullivan-imanage commented on src/main/java/com/example/demo/services/BookingService.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    public void deleteBooking(Long id) {
+        BookingEntity booking = bookingEntityRepository.findById(id).get();
```
I think this will go badly if the booking didn't exist - i.e. you can't do a get on an optional that has isPresent() returning null. This means you need to have a test that tests these methods for error cases, such as calling deleteBooking for an id you know is not in the DB.

SamuelConnor committed:
Updated issues with too many repo calls.
Added tests for deleting a booking that doesn't exist, and updating a booking that doesn't exist.
